### PR TITLE
Add twig rendering for gift card configuration template

### LINKED ISSUE #210: Use twig template directly in gift card configuration
It would be more flexible to the administrator if it was possible to edit a twig template directly when updating the gift card configuration instead of having to resort to CSS and updating the template in the code base

image

### PR DESCRIPTION
**Pull request by loevgaard**
Fixes #210 

This PR will allow the user to edit the gift card template as twig instead of only being able to change the CSS.

- [ ] Tests
- [ ] Allow user to load default twig content
- [ ] Validate twig content with a sandboxed environment before previewing and saving
- [ ] Update fixtures

### CODE REVIEW

File: src/Renderer/GiftCardRendererInterface.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+use Setono\SyliusGiftCardPlugin\Model\GiftCardInterface;
+
+interface GiftCardRendererInterface
+{
+    /**
+     * A boolean indicating whether the rendered PDF will be base64 encoded or not
+     *
+     * Default: false
+     */
+    public const OPTION_BASE_64 = 'base64';
```
Comment (Roshyo) [inline src/Renderer/GiftCardRendererInterface.php]: Is that really a boolean ? I mean, it looks like a string right now :thinking: :laughing:

Comment (loevgaard) [reply src/Renderer/GiftCardRendererInterface.php]: Didn't know how to document it :D The option when sent to the `GiftCardRendererInterface` is a boolean :D

File: src/Renderer/GiftCardPDFRenderer.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    private NormalizerInterface $normalizer;
+
+    public function __construct(
+        Environment $twig,
+        GiftCardConfigurationProviderInterface $configurationProvider,
+        ChannelContextInterface $channelContext,
+        LocaleContextInterface $localeContext,
+        GeneratorInterface $snappy,
+        PdfRenderingOptionsProviderInterface $renderingOptionsProvider,
```
Comment (loevgaard) [inline src/Renderer/GiftCardPDFRenderer.php]: `PDF` should be either named `Pdf` or `PDF`. Choose one

File: src/Api/CommandHandler/CreateGiftCardConfigurationHandler.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            $giftCardConfiguration->setPageSize($command->pageSize);
+        }
+        if (null !== $command->orientation) {
+            $giftCardConfiguration->setOrientation($command->orientation);
+        }
+        if (null !== $command->template) {
+            $giftCardConfiguration->setTemplate($command->template);
+        }
+
+        $this->giftCardConfigurationManager->persist($giftCardConfiguration);
```
Comment (loevgaard) [inline src/Api/CommandHandler/CreateGiftCardConfigurationHandler.php:47]: @Roshyo: Is it normal to persist here?

File: src/Renderer/PDFResponse.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Renderer;
+
+use Symfony\Component\HttpFoundation\HeaderUtils;
+use Symfony\Component\HttpFoundation\Response;
+
+final class PDFResponse
```
Comment (loevgaard) [inline src/Renderer/PDFResponse.php]: Should this be renamed to just `PDF`?

**Pull request merged**